zebra/zebra_cli.c

```diff
@@ -230,11 +230,12 @@ DEFUN_YANG_NOSH (link_params,
 
 	ret = nb_cli_apply_changes(vty, NULL);
 	if (ret == CMD_SUCCESS) {
-		char xpath[XPATH_MAXLEN];
+		char *xpath;
```

Comment:
if we're going to touch this now, can we move the declaration to the beginning of the function?

Reply:
why?

Reply:
FWIW in the past the main justification for the preference for all variables to be declared at the beginning of the function was to track stack usage by the function. 

This was of particular interest inside kernels where you may be working with a limited per-proces kernel stack. But, we aren't a kernel and this isn't even some large array of data (which still might warrant being at the top of the function for same reasons). Rather it's a temporary variable with recognizable limited scope so I actually think it's in the right place as is.

Reply:
I've thought that this was a common FRR practice/convention, so I've often applied it in reviews (and my own FRR code).

Reply:
maybe should say a little more there: I like that the convention helps avoid shadowing problems, and helps avoid endless nitpicking about where exactly declarations should appear (once the box is opened).

Reply:
Chris just keeps the location as-is in this commit, and it's fixing a build failure on newer compiler versions that some people already reported. So I'd like to merge it now and not wait for another CI cycle.

I personally like the current version, because it defines a value in a limited scope, uses it immediately and removes. I think it makes more sense than defining it at the top. Anyway, we can decide on the general convention separately and fix this and other similar definitions separately if needed.